src/backend/distributed/executor/multi_server_executor.c

```diff
@@ -89,13 +112,13 @@ JobExecutorType(DistributedPlan *distributedPlan)
 	taskCount = list_length(job->taskList);
 	tasksPerNode = taskCount / ((double) workerNodeCount);
 
-	if (executorType == MULTI_EXECUTOR_REAL_TIME)
+	if (adaptiveExecutorEnabled || executorType == MULTI_EXECUTOR_REAL_TIME)
```

Comment:
This seems a bit confusing given that most of the if statements below do not trigger when `adaptiveExecutorEnabled` is false.